src/main/kotlin/com/sourcegraph/cody/edit/widget/LensWidgetGroup.kt

```diff
@@ -250,6 +252,39 @@ class LensWidgetGroup(val session: FixupSession, parentComponent: Editor) :
     widgets.forEach { Disposer.register(this, it) }
   }
 
+  // Computes the X coordinate in the Editor where the first widget is drawn.
+  private fun leftMargin(): Int {
+    try {
+      val document = editor.document
+      val inlayOffset = inlay?.offset ?: return DEFAULT_MARGIN
+      val lineCount = document.lineCount
+
+      val inlayLineNumber = document.getLineNumber(inlayOffset)
+
+      // Find next non-blank line.
+      for (lineNumber in inlayLineNumber until lineCount) {
+        val lineStartOffset = document.getLineStartOffset(lineNumber)
+        val lineEndOffset = document.getLineEndOffset(lineNumber)
+        val lineText = document.getText(TextRange(lineStartOffset, lineEndOffset))
+
+        // Compute the pixel width of the indentation.
+        if (lineText.isNotBlank()) {
+          val tabSize = EditorUtil.getTabSize(editor)
+          val spaceWidth = EditorUtil.getSpaceWidth(Font.PLAIN, editor)
+          val indentationLevel =
+              lineText
+                  .takeWhile { it.isWhitespace() }
+                  .sumOf { if (it == '\t') tabSize * spaceWidth else spaceWidth }
+
+          return indentationLevel
```

Comment:
how often this calculations happen? is it on every UI update? we should be careful and avoid heavy operations on AWT (I am not saying it's heavy - just asking). 

@steveyegge, can you verify/confirm it's safe? 🙏

Reply:
Great callout @mkondratek -- it was getting called a lot. I cached it and it's only called when it changes now.